app/src/main/java/uk/ac/ebi/atlas/experimentpage/baseline/topgenes/BaselineExperimentTopGenesService.java

```diff
@@ -37,6 +37,8 @@ public BaselineExperimentTopGenesService(BaselineExperimentTopGenesDao baselineE
                      baselineExperimentTopGenesDao.aggregateGeneIdsAndSortBySpecificity(
                              experimentAccession, preferences)) {
             return mapBySpecifictyAndSortByAverageExpression(tupleStreamer.get(), preferences.getHeatmapMatrixSize());
+        } catch (Exception e) {
+            throw new RuntimeException("The Expression Atlas Solr server could not be reached.");
```

Comment:
If this is what it is going to appear on the UI, please add some advice for the user to try to recover from the error. In most cases I find that a second request (usually with a hard reload) some minutes later, when the previous failed query is probably cached in solr, works. Same for the other message please.